src/backend/distributed/planner/multi_join_order.c

```diff
@@ -823,12 +823,13 @@ ReferenceJoin(JoinOrderNode *currentJoinNode, TableEntry *candidateTable,
 		return NULL;
 	}
 
-	char candidatePartitionMethod = PartitionMethod(candidateTable->relationId);
-	char leftPartitionMethod = PartitionMethod(currentJoinNode->tableEntry->relationId);
-
 	if (!IsSupportedReferenceJoin(joinType,
-								  leftPartitionMethod == DISTRIBUTE_BY_NONE,
-								  candidatePartitionMethod == DISTRIBUTE_BY_NONE))
+								  IsCitusTableType(
+									  currentJoinNode->tableEntry->relationId,
```

Comment:
could use intermediate variables, pretty hard to follow
```
bool leftIsReferenceTable = ....
bool rightIsReference = ...
```
as these are the parameters that `IsSupportedReferenceJoin ` gets